pkg/lightning/restore/table_restore.go

```diff
@@ -694,15 +693,28 @@ func (tr *TableRestore) postProcess(
 					return false, err
 				}
 				if !needChecksum {
+					if rc.cfg.TikvImporter.DuplicateDetection {
+						if err := rc.backend.CollectLocalDuplicateRows(ctx, tr.encTable, rc.cfg.TiDB.SQLMode); err != nil {
+							tr.logger.Error("collect local duplicate keys failed", log.ShortError(err))
+						}
+					}
 					return false, nil
 				}
+				if rc.cfg.TikvImporter.DuplicateDetection {
+					if err := rc.backend.CollectRemoteDuplicateRows(ctx, tr.encTable, rc.cfg.TiDB.SQLMode); err != nil {
+						tr.logger.Error("collect remote duplicate keys failed", log.ShortError(err))
+						err = nil
+					}
+				}
 				if cp.Checksum.SumKVS() > 0 || baseTotalChecksum.SumKVS() > 0 {
 					localChecksum.Add(&cp.Checksum)
 					localChecksum.Add(baseTotalChecksum)
 					tr.logger.Info("merged local checksum", zap.Object("checksum", &localChecksum))
 				}
 
-				err = tr.compareChecksum(ctx, localChecksum)
+				remoteChecksum, err := DoChecksum(ctx, tr.tableInfo)
+				// TODO: If there are duplicate keys, do not set the `ChecksumMismatch` error
```

Comment:
Yes. So I left a comment `TODO`.  Because maybe we will store the duplicate keys in TiDB, so there may be other method to judge whether we need to checksum/

Reply:
Okay